### PR TITLE
convert translation to gettext style

### PR DESCRIPTION
**Pull request by restena-sw**
script which shuffles .po file content from SSP-style codes to normal gettext() style msgid = EN, msgstr = XX language files, and back again

### CODE REVIEW

**Comment by tvdijen** [comment]
I'm having a hard time understanding what this script does.. Can you give an example?

**Comment by restena-sw** [comment]
Hi,

the translations in the locale/ subdirs contain entries like

en/   

msgid "{logout:failed}"
msgstr "Logout failed"

fr/
msgid "{logout:failed}"
msgstr "quelque chose"

That structure is entirely unhelpful for CAT tools as they put the English source (not an abstract string) and the desired translation side-by-side, as in:

msgid "Logout failed"
msgstr "quelque chose"

As you see, this involves not much more than inverting orders of strings and changing msgstr to msgid.

One of the operation modes of the script is to do just that - take two individual language input files and generate one which has English as msgid and the translation as msgstr.

Outputs of that do not contain the original "glueing" abstract string any more - which is what translators want. But eventually, after working with the generated file, the translation needs to be brought back into SSPs native format.

That's why the inverse operation mode takes a file from original sources (abstract string+English) and generates output with the abstract string and the target translation instead. Tada!

The code is rather dense, admittedly. That's because it mostly does some s/msgstr/msgid and at other places juggles line indexes around.

**Comment by restena-sw** [comment]
For context, you may also want to read my mailing list post from 04 March 2020 "Translation remarks / question".

File: bin/convertTranslations.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+");
+    exit(1);
+}
+
+switch ($argv[1]) {
+    case "MERGE":
+        fwrite(STDERR, "Will merge two language into one .po file based on identical msgIds.\n");
+        break;
+    case "CODIFY":
+        fwrite(STDERR, "Will create .po file with codes as msgIds and dest langauge translations as msgStr.\n");
```
Comment (tvdijen) [inline bin/convertTranslations.php:86]: typo: language

**Comment by tvdijen** [comment]
Ah ok.. I always use the dictionaryconverter script for that..

**Comment by restena-sw** [comment]
"replace the tag-codes with actual sentences" sounds like manual work is needed every time? That's where an automation script helps :-)
Also, I had the impression that JSON dictionaries are deprecated/obsolete so that won't be possible for long?

**Comment by tvdijen** [comment]
Yes, that's a manual action, but it's just in one file instead of every PO-file..
The JSON-dictionaries are already gone on `master`, but of course we still have them in the 1.19 branch..

**Review by thijskh — APPROVED** [review]
Just what we need to clean up the old translation styles. Thanks!

**Pull request merged**